blockdevice/blockdevice.go

```diff
@@ -101,6 +101,9 @@ type BlockDevice struct {
 	// DeviceType is the type of the blockdevice. can be sparse/disk/partition etc
 	DeviceType string
 
+	// DriveType is the type of backing drive for this blockdevice. HDD/SSD/sparse
+	DriveType string
```

Comment:
@akhilerm what is the JSON path for DriveType and DeviceType in block device?

Reply:
This is the `BlockDevice` struct that will be used internally. We are deprecating the `DiskInfo` and `DeviceInfo` struct in favour of this `BlockDevice` struct which will form the core. 

The JSON path is defined for the `BlockDevice` custom resource, `.spec.details.deviceType`.
There is no drive type field currently in `BlockDevice` resource.

Reply:
@akhilerm do we need to update API struct also?

Reply:
This PR contains changes to the internal struct only. Changes to the API and its uses can be updated in further PRs. 
I would prefer it to be done, along with the sysfs probe where logical size, type of drive etc can be updated.